### ISSUE TITLE
Should not allow constructing instances of abstract types

### ISSUE THREAD

**Issue by jeromesimeon** [accordproject/ergo#370]
CTO types can be abstract. E.g., `abstract asset AccordContract identified by contractId` the type checker should prevent constructions of objects when the type is abstract.
- [x] Add abstract markers in the CTO and ErgoType ASTs
- [x] Add abstract syntax for Ergo types
- [x] Make creation of abstract types a type error

**Comment by jeromesimeon** [comment]
Looks better already:
```
$ cat ~/test.cto 
namespace org.accordproject.forest

abstract concept Animal {
}

concept Bear extends Animal{
  o String name
  o Integer age
}
$ ./bin/ergotop ~/test.cto 
Welcome to ERGOTOP version 0.3.4
ergo$ import org.accordproject.forest.*
ergo$ return Bear{ age : 1, name: "winnie" };
Response. Bear{age: 1, name: "winnie"} : Bear
ergo$ return Animal{};
CompilationError (at line 1 col 7). Cannot create instance of abstract type `org.accordproject.forest.Animal'
return Animal{};
       ^^^^^^^^ 
ergo$ 
```

**Comment by jeromesimeon** [comment]
And also:
```
$ cat ~/test.ergo
namespace org.accordproject.forest

define abstract concept Being{}

define concept Animal extends Being{
  name : String,
  age : Integer
}
define concept Rabbit extends Animal{
}
define concept Bear extends Animal{
  food : String
}
define concept Raccoon extends Rabbit{
  weapon : String
}
$ ./bin/ergotop ~/test.ergo 
Welcome to ERGOTOP version 0.3.4
ergo$ import org.accordproject.forest.*
ergo$ return Animal{ age : 1, name: "winnie" };
Response. Animal{age: 1, name: "winnie"} : Animal
ergo$ return Being{};
CompilationError (at line 1 col 7). Cannot create instance of abstract type `org.accordproject.forest.Being'
return Being{};
       ^^^^^^^ 
ergo$ 
```

**Comment by jeromesimeon** [comment]
Fixed in #434
